crates/axvmhal/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+
+#![no_std]
+use core::marker::Sized;
+
+pub use memory_addr::{PhysAddr, VirtAddr};
+/// Guest virtual address.
+pub type GuestVirtAddr = usize;
+/// Guest physical address.
+pub type GuestPhysAddr = usize;
+/// Host virtual address.
+pub type HostVirtAddr = VirtAddr;
+/// Host physical address.
+pub type HostPhysAddr = PhysAddr;
+pub use memory_addr::PAGE_SIZE_4K as PAGE_SIZE;
+
+/// The interfaces which the underlying software (kernel or hypervisor) must implement.
+
+ pub trait AxVMHal: Sized + Clone {
```

Comment:
After a discussion with @equation314 (he suggested  that [crate_interface](https://crates.io/crates/crate_interface) can be used here). 

Placing all these interfaces into a single trait is not a good design. 
It's better to use a super trait or an associated type， @aarkegz 

e.g.
`PagingHandler` for [axaddrspace](https://github.com/arceos-hypervisor/axaddrspace)
`VMMHandler` for semantics like `get_vm_by_id()`

Reply:
> Placing all these interfaces into a single trait is not a good design.
> It's better to use a super trait or an associated type

I agree.



> he suggested that [crate_interface](https://crates.io/crates/crate_interface) can be used here

`crate_interface` is a good solution.

Reply:
> `crate_interface` is a good solution.

But `crate_interface` cannot support super trait or an associated type